Docker/CI.Ubuntu22.dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:22.04
+
+# basic dependencies -
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -y && \
+  apt-get install -y 
+
+RUN apt-get install -y curl \
+  pkg-config \
+  findutils \
+  wget
+
+# build dependencies
+RUN apt-get install -y && \
+    apt-get remove -y llvm-12 clang-format-11 clang-tidy-11 && \
+    apt-get install -y curl pkg-config findutils  wget && \
+    apt install -y clang-13 cmake clang-format-13 clang-tidy-13 && \
```

Comment:
(nit)
I assume that some future version of Ubunto is allowed to not contain the `llvm-12` or `clang-format-11` or `clang-tidy-11`, listed in L15, that is why the command "apt-get **remove**" (L15) can _fail_, which, combined with the `&&` operator in the end of the L15, can result in _skipping the commands in L16+_.

Consider executing the command in L15 _separately_. And _regardless of its success_ execute the commands in L14, L16+.

Reply:
When building a docker image, it is common practice to concatenating lines using `&&`. The reason for this is that every command `RUN`/`ADD`/etc. creates a new cache that potentially takes up a lot of space on your hard drive. I agree that this ruins debuggability, but unfortunately not doing this will exhaust the hard drive quickly.